app/models/episode.rb

```diff
@@ -19,10 +19,12 @@ def update(args)
 
   class << self
     def prepare_params(args)
-      raise(ActiveRecord::RecordInvalid.new, 'Invalid video') unless args[:video].match VALID_YOUTUBE_LINK
+      if args[:video] && args[:date]
+        raise(ActiveRecord::RecordInvalid.new, 'Invalid video') unless args[:video].match VALID_YOUTUBE_LINK
 
-      args[:video] = args[:video].match(YOUTUBE_VIDEO_IDENTIFIER)[5]
-      args[:date] = Time.parse(args[:date]).utc
+        args[:video] = args[:video].match(YOUTUBE_VIDEO_IDENTIFIER)[5]
```

Comment:
O_o How can I miss this =\
We will fully rewrite this model later. Sorry
Don't understand why we need this `self` methods.
You can define `update_youtube` or some staff like this and validate url only on this method

Feel free to resolve this one